### PR TITLE
Fix asset filtering

### PR DESCRIPTION
**Pull request by robbie-daniels**
## What does this PR do
* Fixes the issue displaying the asset page for wallets with >10 assets
* Adds a test to make sure that assets are successfully displayed, when there is a mis-match between what's on the ledger and what's in firebase.

## How should this PR be tested?
- [ ] Log in with a wallet that has >10 assets in firebase, and load the asset page.  Confirm that it loads successfully

## Checklist before requesting a review
- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by robbie-daniels** [comment]
* The first commit on this PR is misleadingly named - it's just a generic catch-all for any other errors we might get on the asset screen
* Open to proposals on alternative behaviour, for cases when there is a mis-match between what's in firestore vs what's on the ledger - will update the test case accordingly

File: lib/marketplace/marketplace_home_screen.dart [hunk 1]
```diff
@@ -182,18 +184,50 @@ class _AssetScreenState extends State<MarketplaceHomeScreen>
...
+                    List<Stream<QuerySnapshot<Map<String, dynamic>>>> streams =
+                        [];
+                    for (int i = 0; i < nfTokenIDs.length; i += 10) {
+                      List<dynamic> chunk = nfTokenIDs.sublist(
+                          i,
+                          i + 10 > nfTokenIDs.length
+                              ? nfTokenIDs.length
+                              : i + 10);
+
+                      // TODO: This limits to 20 - needs to handle cases with more than 20:
```
Comment (desimmons) [inline lib/marketplace/marketplace_home_screen.dart:205]: It's not clear what this TODO relates to

**Review by desimmons — APPROVED** [review]
I think this approach looks good!

File: lib/marketplace/marketplace_home_screen.dart [hunk 2]
```diff
@@ -209,58 +249,80 @@ class _AssetScreenState extends State<MarketplaceHomeScreen>
...
+      padding: const EdgeInsets.only(top: 8),
+      scrollDirection: Axis.vertical,
+      gridDelegate: SliverGridDelegateWithMaxCrossAxisExtent(
+          maxCrossAxisExtent: 600, childAspectRatio: 3),
+      itemBuilder: (BuildContext context, int index) {
+        final int count = docs.length > 10 ? 10 : docs.length;
+        final Animation<double> animation = Tween<double>(begin: 0.0, end: 1.0)
+            .animate(CurvedAnimation(
+                parent: animationController!,
+                curve: Interval(min((1 / count) * index, 1.0), 1.0,
```
Comment (desimmons) [inline lib/marketplace/marketplace_home_screen.dart:298]: @robbie-daniels the "begin must be <= 1.0" was resolved by adding the `min` here.

**Pull request merged**